curp/src/server/raw_curp/mod.rs

```diff
@@ -233,72 +233,63 @@ impl<C: 'static + Command, RC: RoleChange + 'static> RawCurp<C, RC> {
     pub(super) fn handle_propose(
         &self,
         cmd: Arc<C>,
-    ) -> ((Option<ServerId>, u64), Result<bool, ProposeError>) {
+    ) -> Result<((Option<ServerId>, u64), Result<bool, ProposeError>), CurpError> {
```

Comment:
A weird Result type nest a Result type. Maybe we could find another design here...

Reply:
It's not that weird. The `CurpError` is an out-of-bound error while the other is an inner error. And the inner error will be wrapped into the ProposeResponse finally.